lib/jitter/sleep.go

```diff
@@ -7,19 +7,26 @@ import (
 
 const DefaultMaxMs = 3500
 
-func Jitter(baseMs, maxMs, attempts int) time.Duration {
+func computeJitterUpperBoundMs(baseMs, maxMs, attempts int) int {
 	// https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
 	// sleep = random_between(0, min(cap, base * 2 ** attempt))
 	// 2 ** x == 1 << x
 	if maxMs <= 0 {
-		return time.Duration(0)
+		return 0
 	}
 
 	// Check for overflows when computing base * 2 ** attempts.
 	if attemptsMaxMs := baseMs * (1 << attempts); attemptsMaxMs > 0 {
 		maxMs = min(maxMs, attemptsMaxMs)
 	}
 
-	ms := rand.Intn(maxMs)
-	return time.Duration(ms) * time.Millisecond
+	return maxMs
+}
+
+func Jitter(baseMs, maxMs, attempts int) time.Duration {
+	upperBoundMs := computeJitterUpperBoundMs(baseMs, maxMs, attempts)
+	if upperBoundMs <= 0 {
+		return time.Duration(0)
```

Comment:
can we just return max ms here?

Reply:
I'm like 95% sure here that if we've made it to this line `maxMs` will be <= 0, but explicitly returning `time.Duration(0)` lets us avoid returning negative numbers if `maxMs` is negative.